### PR TITLE
[DB TimeLock] 1D - Agent Wiring

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**:
- DB Timelock needs to support multiple series

**Implementation Description (bullets)**:
- Set db timelock up to use the new multi series bound store.
- Pass the series in as a client.

**Testing (What was existing testing like?  What have you done to improve it?)**:
The ETEs validate that nothing has truly imploded. There are some small unit tests. More involved testing should be handled in task 3 of this workstream.

**Concerns (what feedback would you like?)**:
- `DbTimestampCreationParameters` would really better be served by a Visitor pattern and subtypes, though I think that's a bit too much work and not particularly valuable. What do you think?
- Some duplication in the checks for Cassandra and JDBC. I will separately check if we can kill the latter, this is frustrating. Impl-Shared is strangely not pulled in by JDBC and there aren't many choices on where
- The testing setup for this PR is admittedly limited, and I do not like it. That said, we will not roll this out until it's further improved, and this code path is not used in production anywhere, and I think this component is reviewable by itself.

**Where should we start reviewing?**: DbBoundTimestampCreator is the entrypoint and probably where you should drill down from.

**Priority (whenever / two weeks / yesterday)**: this week for a first pass (though not the highest priority as I'll be back in on Monday)

### CODE REVIEW

File: atlasdb-api/src/main/java/com/palantir/atlasdb/config/DbTimestampCreationSetting.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@Data
+public abstract class DbTimestampCreationSetting {
+    public interface Cases<R> {
+        R multipleSeries(Optional<TableReference> tableReference, TimestampSeries series);
+        R singleSeries(Optional<TableReference> tableReference);
+    }
+
+    public abstract <R> R match(Cases<R> cases);
+
+    @Override
```
Comment (gmaretic) [inline atlasdb-api/src/main/java/com/palantir/atlasdb/config/DbTimestampCreationSetting.java:35]: Note: these indicate that the methods should be generated

File: timelock-agent/src/main/java/com/palantir/timelock/paxos/DbBoundTimestampCreator.java [hunk 2]
```diff
@@ -49,15 +52,18 @@ public DbBoundTimestampCreator(KeyValueServiceConfig kvsConfig) {
...
-        Preconditions.checkArgument(timestampService instanceof TimestampManagementService,
-                "The timestamp service is not a managed timestamp service.");
 
         return () -> new DelegatingManagedTimestampService(timestampService,
                 (TimestampManagementService) timestampService);
     }
+
+    @VisibleForTesting
+    static DbTimestampCreationSetting getTimestampCreationParameters(Client client) {
+        return DbTimestampCreationSettings.multipleSeries(Optional.empty(), TimestampSeries.of(client.value()));
```
Comment (gmaretic) [inline timelock-agent/src/main/java/com/palantir/timelock/paxos/DbBoundTimestampCreator.java:67]: Is it safe to just switch here?

**Review by gmaretic — APPROVED** [review]
LGTM, just have one question that is really relevant. Otherwise double check that you agree with my changes and ship

File: atlasdb-api/src/main/java/com/palantir/atlasdb/config/DbTimestampCreationSetting.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/config/DbTimestampCreationSetting.java:35]: interesting - I guess you wouldn't want them if you aren't testing or putting them in sets, though it does feel a bit awkward that these aren't by default

File: timelock-agent/src/main/java/com/palantir/timelock/paxos/DbBoundTimestampCreator.java (hunk 2 above)
Comment (jeremyk-91) [reply timelock-agent/src/main/java/com/palantir/timelock/paxos/DbBoundTimestampCreator.java:67]: Yes. This isn't currently used in production anywhere.

**Comment by jeremyk-91** [comment]
👍

**Pull request merged**